components/omega/doc/design/Broadcast.md

```diff
@@ -0,0 +1,132 @@
+<!--- OMEGA Broadcast Requirements and Design --------------------------------->
+
+# OMEGA Requirements and Design:
+## *Broadcast*
+
+## 1 Overview
+
+In a parallel environment, it is often useful to broadcast values from
+one MPI Rank to the others. In particular, the current design for
+OMEGA configuration requires us to broadcast many aspects of that
+configuration from the master rank to all others.
+
+## 2 Requirements
+
+### 2.1 Requirement: Broadcast all supported data types
+
+We must be able to broadcast variables of I4, I8, R4, R8, Real,
+boolean, std::string. 
+
+### 2.2 Requirement: Broadcast scalars and vectors
+
+Generally we will be broadcasting scalar values, but will require
+vectors of values in some cases. 
+
+### 2.3 Requirement: Broadcast from any rank
+
+The most common use case will broadcast from the master rank but
+we must be able to support broadcasting from any rank.
+
+### 2.4 Desired: Broadcast within alternative environments
+
+OMEGA plans to support subset partitions of the domain for sub-models,
+so we must be able to broadcast only within these sub-groups using
+the appropriate communicator.
+
+## 3 Algorithmic Formulation
+
+No algorithms needed beyond the MPI broadcast internal algorithm.
+
+## 4 Design
+
+The OMEGA broadcast is mostly just wrappers around the MPI broadcast
```

Comment:
Thinking out loud: We should facilitate nonblocking broadcasts as an option in the framework. 
Example: https://www.mpi-forum.org/docs/mpi-3.1/mpi31-report/node128.htm#Node128

Reply:
That's a good point. It'll likely be more performant, though I think we've run into some issues in the past when doing too many non-blocking messages when we're broadcasting a lot during init. I'll add an option for it anyway.